src/pdf_parser/pdf_utils/ocr.py

```diff
@@ -1,11 +1,329 @@
+                    for symbol in word.symbols:
+                        line += symbol.text
+                        break_type = symbol.property.detected_break.type
+                        # Add space to end of line if it's not a break.
+                        if break_type in [breaks.SPACE, breaks.EOL_SURE_SPACE]:
+                            line += " "
+                        # Start new line in same paragraph if there is a line break or a sure space (i.e. large space)
+                        if break_type in [breaks.LINE_BREAK, breaks.EOL_SURE_SPACE]:
+                            lines.append(line)
+                            para += line
+                            line = ""
+                # Detect language by selecting the modal language of the words in the paragraph.
+                para_lang = _get_modal_string(para_languages)
+                paragraph_text_segments.append(
+                    GoogleTextSegment(
+                        text=para,
+                        coordinates=paragraph.bounding_box,
+                        confidence=paragraph.confidence,
+                        language=para_lang,
+                    )
+                )
+                default_dict["block_paragraphs"].append(para)
+                default_dict["block_paragraph_coords"].append(paragraph.bounding_box)
+
+            # Detect language by selecting the modal language of the words in the block.
+            block_lang = _get_modal_string(default_dict["block_languages"])
+            # for every block, create a text block
+            block_all_text = "\n".join(default_dict["block_paragraphs"])
+            block_text_segments.append(
+                GoogleTextSegment(
+                    coordinates=block.bounding_box,
+                    text=block_all_text,
+                    confidence=block.confidence,
+                    language=block_lang,
+                )
+            )
+
+            # For every block, create a block object (contains paragraph metadata).
+            block_list = [
+                GoogleTextSegment(
+                    coordinates=default_dict["block_paragraph_coords"][i],
+                    text=default_dict["block_paragraphs"][i],
+                    language=default_dict["block_languages"][i],
+                    confidence=default_dict["paragraph_confidences"][i],
+                )
+                for i in range(len(default_dict["block_paragraphs"]))
+            ]
+            google_block = GoogleBlock(
+                coordinates=block.bounding_box, text_blocks=block_list
+            )
+            fully_structured_blocks.append(google_block)
+
+    # look for duplicates in block_texts and paragraph_texts and create a list of full blocks
+    text_blocks_to_keep = []
+    for block in block_text_segments:
+        if block in paragraph_text_segments:
+            continue
+        else:
+            text_blocks_to_keep.append(block)
```

Comment:
text_blocks_to_keep = [block for block in block_text_segments if block not in paragragh_text_segments]